ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -462,6 +475,44 @@ private void expandNestedListType(String jsonFieldName, Object val, Field field,
     });
   }
 
+  /**
+   * This method gives us a chance to perform some additional type coercions in case the BigQuery
+   * field type is different from the source data type. This should rarely happen, since only
+   * validated payloads get through to this BQ sink path, but there are sets of probes with
+   * heterogeneous types that appear as explicit fields in BQ, but are treated as loosely typed
+   * maps at the validation phase; we need to catch these or they can cause the entire pipeline
+   * to stall.
+   *
+   * <p>Returning {@code null} here indicates that no coercion is defined and that the field should
+   * be put to {@code additional_properties}.
+   */
+  private static Object coerceToBqType(Object o, Field field) {
+    if (field.getMode() == Mode.REPEATED) {
+      return o;
+    } else {
+      if (field.getType() == LegacySQLTypeName.STRING) {
```

Comment:
this condition will never evaluate as true when called from `processField`

Reply:
You are correct. I was attempting to change the existing code as little as possible, but it should be possible to remove one of the cases in `processField`.

Really, I should remove a few of the cases and handle coercing `List`s as well. I think I'll go ahead and make those changes.

Reply:
see also my other comment about whether this shouldn't be in a separate function